static/js/metrics-explorer.js

```diff
@@ -238,6 +256,7 @@ async function initializeFormulaFunction(formulaElement, uniqueId) {
         // If the validation passes, call the getMetricsDataForFormula with the updated details
         if (validationResult !== false) {
             await getMetricsDataForFormula(uniqueId, validationResult);
+            updateMetricsQueryParamsInUrl();
```

Comment:
The `getQueryDetails` function is called every time we change the query/formula, and when the time changes, we call `refreshMetricsGraphs`. Instead of adding the `updateMetricsQueryParamsInUrl` function in multiple places, you can add it directly to the `getQueryDetails` and  `refreshMetricsGraphs` functions.